### PR TITLE
[l10n] Improve Hebrew (he-IL) locale

### PR DESCRIPTION
**Pull request by itayG98**
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/mui-x/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

- [x] I have added the missing Hebrew translation since I noticed that Hebrew support was not available in the [docs](https://mui.com/x/react-data-grid/localization/).

Your effort is much appreciated, and we look forward to reviewing your changes. If there are any questions or concerns, feel free to let us know.

Thank you again for your contribution! 🙏

### CODE REVIEW

File: docs/data/date-pickers/localization/data.json
Comment (itayG98) [inline docs/data/date-pickers/localization/data.json:1]: Ran yarn prettier

File: docs/data/data-grid/localization/data.json
Comment (itayG98) [inline docs/data/data-grid/localization/data.json:1]: Ran yarn prettier

**Comment by itayG98** [comment]
I cant see why the test fails with "`yarn l10n` changes committed?" error

**Comment by LukasTy** [comment]
> I cant see why the test fails with "`yarn l10n` changes committed?" error

Because the environment that you ran `yarn l10n` generated incorrect paths. 🙈 
The `/` has been replaced with `\\`.
Are you on Windows?
Maybe running the script on WSL would avoid this issue? 🤔

**Comment by itayG98** [comment]
> > I cant see why the test fails with "`yarn l10n` changes committed?" error
> 
> Because the environment that you ran `yarn l10n` generated incorrect paths. 🙈 The `/` has been replaced with `\\`. Are you on Windows? Maybe running the script on WSL would avoid this issue? 🤔

I see now the diffrence
I will handle this soon

**Review by LukasTy — APPROVED** [review]
Thank you for your contribution and taking the time to finalize and clean up the PR. 🙏 💯

**Comment by itayG98** [comment]
> Thank you for your contribution and taking the time to finalize and clean up the PR. 🙏 💯

You welcome !!

**Pull request merged**